redash/query_runner/python.py

```diff
@@ -187,9 +184,9 @@ def run_query(self, query):
             #       One option is to use ETA with Celery + timeouts on workers
             #       And replacement of worker process every X requests handled.
 
-            exec(code) in restricted_globals, script_locals
+            exec(code) in restricted_globals, self._script_locals
 
-            json_data = json.dumps(self._result)
```

Comment:
OK but in that case it may override anything that was set in result["log"]

On Wed, Jul 22, 2015, 17:47 Arik Fraimovich <notifications@github.com>
wrote:

> In redash/query_runner/python.py
> <https://github.com/EverythingMe/redash/pull/503#discussion_r35220733>:
>
> >
> > -            json_data = json.dumps(self._result)
>
> Actually it's better to accumulate the log statements in another variable,
> and then put them in result["log"], or they might overriden -- I'll do it
> in this branch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EverythingMe/redash/pull/503/files#r35220733>.
>